### PR TITLE
chore: Add type casts reducing the diff for #3697

### PR DESCRIPTION
**Pull request by kevaundray**
This adds type casts which are assumed to be benign -- reducing the diff for #3697

### CODE REVIEW

File: specs/_features/eip7594/polynomial-commitments-sampling.md [hunk 1]
```diff
@@ -657,7 +657,7 @@ def recover_original_data(eval_shifted_extended_evaluation: Sequence[BLSFieldEle
 
     reconstructed_data = bit_reversal_permutation(fft_field(reconstructed_poly, roots_of_unity_extended))
 
-    return reconstructed_data
+    return CosetEvals(reconstructed_data)
```
Comment (asn-d6) [inline specs/_features/eip7594/polynomial-commitments-sampling.md:660]: CosetEvals is just 64 field elements. `reconstructed_data` here is likely the full extended length 8192.

Comment (kevaundray) [reply specs/_features/eip7594/polynomial-commitments-sampling.md:660]: Yeah you are right -- this was diffed over from 3697 so it should be an issue there too

**Review by asn-d6 — APPROVED** [review]
LGTM!

**Review by jtraglia — APPROVED** [review]
Nice, this looks good 😄

**Pull request merged**